SWLOR.Game.Server/Feature/AbilityDefinition/OneHanded/ShieldBashAbilityDefinition.cs

```diff
@@ -0,0 +1,106 @@
+﻿//using Random = SWLOR.Game.Server.Service.Random;
+
+using System.Collections.Generic;
+using SWLOR.Game.Server.Core;
+using SWLOR.Game.Server.Core.NWScript.Enum;
+using SWLOR.Game.Server.Service;
+using SWLOR.Game.Server.Service.AbilityService;
+using SWLOR.Game.Server.Service.CombatService;
+using SWLOR.Game.Server.Service.PerkService;
+using SWLOR.Game.Server.Service.SkillService;
+
+namespace SWLOR.Game.Server.Feature.AbilityDefinition.OneHanded
+{
+    public class ShieldBashAbilityDefinition : IAbilityListDefinition
+    {
+        public Dictionary<FeatType, AbilityDetail> BuildAbilities()
+        {
+            var builder = new AbilityBuilder();
+            ShieldBash1(builder);
+            ShieldBash2(builder);
+            ShieldBash3(builder);
+
+            return builder.Build();
+        }
+
+        private static string Validation(uint activator, uint target, int level, Location targetLocation)
+        {
+            var weapon = GetItemInSlot(InventorySlot.LeftHand, activator);
+            var leftHandType = GetBaseItemType(weapon);
+            
+            if (Item.ShieldBaseItemTypes.Contains(leftHandType))
+            {
+                return string.Empty;
+            }
+            else
+                return "A shield must be equipped in your left hand to use this ability.";
+        }
+
+        private static void ImpactAction(uint activator, uint target, int level, Location targetLocation)
+        {
+            // If activator is in stealth mode, force them out of stealth mode.
+            if (GetActionMode(activator, ActionMode.Stealth) == true)
+                SetActionMode(activator, ActionMode.Stealth, false);
+
+            var dmg = 8 * level; // 8, 16, 24 DMG
+            var effDuration = 3f + 3 * level; // 6, 9, 12 sec
+
+            dmg += Combat.GetAbilityDamageBonus(activator, SkillType.OneHanded);
+
+            CombatPoint.AddCombatPoint(activator, target, SkillType.OneHanded, 3);
+
+            var might = GetAbilityModifier(AbilityType.Might, activator);
```

Comment:
Why are you using the modifier instead of the raw score for this damage calculation? Intentional or mistake?